pxtlib/gallery.ts

```diff
@@ -85,6 +85,68 @@ namespace pxt.gallery {
         return prj;
     }
 
+    export function parseCodeCards(md: string): pxt.CodeCard[] {
+        // try to parse code cards as JSON
+        let cards = Util.jsonTryParse(md) as pxt.CodeCard[];
+        if (cards && !Array.isArray(cards))
+            cards = [cards];
+        if (cards?.length)
+            return cards;
+
+        // not json, try parsing as sequence of key,value pairs, with line splits
+        cards = md.split(/^---$/gm)
+            .filter(cmd => !!cmd)
+            .map(cmd => {
+                let cc: any = {};
+                cmd.replace(/^\s*\*\s+(\w+)\s*:\s*(.*)$/gm, (m, n, v) => {
+                    if (n == "flags")
+                        cc[n] = v.split(',')
+                    else
```

Comment:
do we also plan to support arrays of objects like "CodeCardAction" in this format?

Reply:
good added.